src/middlewares/enricher.ts

```diff
@@ -0,0 +1,52 @@
+import { Context, Next } from 'koa';
+import {
+  ProcessorTransformationRequest,
+  RouterTransformationRequest,
+  RouterTransformationRequestData,
+} from '../types';
+import GeoLocationHelper from '../helpers/geoLocation';
+
+export type DTRequest = RouterTransformationRequest | ProcessorTransformationRequest[];
+
+export default class Enricher {
+  private static enrichWithGeoInfo(
```

Comment:
We need to think about how do we handle it, should we do it at destType level or can we go ahead with a more generic implementation.
Rather than complicating for this feature with such thinking, wanted to first get this out to production & may be with one more requirement, there will be more clarity on how best we can handle such kind of things, design them accordingly was the idea.

Let me know if it makes sense

Reply:
It makes sense, let's park it to revisit this part when requirements expand in future.

Reply:
Resolving this comment for now.